app/lib/Controller/AppController.php

```diff
@@ -49,11 +49,16 @@ class AppController {
 	private $opo_dispatcher;
 	
 	# -------------------------------------------------------
-	public static function getInstance(&$po_request=null, &$po_response=null, $dont_create=false) {
+	public static function getInstance(&$po_request=null, &$po_response=null, $dont_create=false, $force_create=false) {
 		global $g_app_controller;
+		if ($force_create) {
+			// Clear old instance, so we create new instance with given request and response.
+			$g_app_controller = null;
```

Comment:
I used `= null` here instead of unsetting it as we immediately set it again in this function. the only scenario where unsetting it would make sense to me is if it was called with both 'dont_create' and 'force_create' where the global would remain unset, but I can't see why you would do that. 

Happy to switch if you think unsetting is better.